File: backend/src/main/java/com/jotoai/voenix/shop/prompts/repository/PromptRepository.java
package com.jotoai.voenix.shop.prompts.repository;

import com.jotoai.voenix.shop.prompts.entity.Prompt;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PromptRepository extends JpaRepository<Prompt, Long> {
    
    List<Prompt> findByTitleContainingIgnoreCase(String title);
}

File: backend/src/main/kotlin/com/jotoai/voenix/shop/prompts/repository/PromptRepository.kt
package com.jotoai.voenix.shop.prompts.repository

import com.jotoai.voenix.shop.prompts.entity.Prompt
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PromptRepository : JpaRepository<Prompt, Long> {
    
    fun findByTitleContainingIgnoreCase(title: String): List<Prompt>
    
    fun countByCategoryId(categoryId: Long): Int
}